### ISSUE TITLE
[Bug] GLM-5.1 TP8 HiCache hangs when device KV available is low but evictable cache is high

### ISSUE THREAD

**Issue by gnatseira** [sgl-project/sglang#26357]
## Description

We are running GLM-5.1 FP8 with TP=8, FP8 KV cache, HiCache, and EAGLE speculative decoding. The service is not stable in long-running production-like runs.

With the default/recommended EAGLE setting `--speculative-num-steps 3 --speculative-eagle-topk 1 --speculative-num-draft-tokens 4`, the service usually only runs for a few hours before failing. After reducing speculative decoding to `1/1/1`, the service becomes more stable and can run for about two days, but it still eventually hangs.

In the latest hang, one TP scheduler stopped making progress and the SGLang scheduler watchdog killed the process. At the time of the hang, the device KV pool had very low `available` tokens but a very large `evictable` cache:

```text
[full] total=464256, available=768, evictable=432960, protected=30464, session_held=0, uncached=0
[2026-05-26 13:36:14 TP5] Scheduler watchdog timeout (self.watchdog_timeout=300, self.soft=False)
[2026-05-26 13:36:19] SIGQUIT received. signum=None, frame=None. It usually means one child failed.
```

Only one request was running according to our monitoring. A single request should not exhaust the whole KV pool; the pool appears mostly occupied by evictable radix/HiCache entries. It looks like a request triggers eviction / HiCache cache-state transition when `available` is very low, but the scheduler gets stuck instead of evicting from the large evictable region.

We have also seen earlier failures in related runs where NCCL collectives timed out in EAGLE/spec-v2 paths. Reducing EAGLE from `3/1/4` to `1/1/1` makes the service much more stable, but does not eliminate the remaining HiCache/radix eviction hang.

## Launch command

```bash
SGLANG_ENABLE_SPEC_V2=1 sglang serve \
  --model-path /path/to/glm-5.1-fp8 \
  --tp 8 \
  --reasoning-parser glm45 \
  --tool-call-parser glm47 \
  --speculative-algorithm EAGLE \
  --speculative-num-steps 1 \
  --speculative-eagle-topk 1 \
  --speculative-num-draft-tokens 1 \
  --mem-fraction-static 0.82 \
  --kv-cache-dtype fp8_e4m3 \
  --enable-hierarchical-cache \
  --hicache-size 80 \
  --host 0.0.0.0 \
  --port 8000 \
  --context-length 160000 \
  --max-running-requests 32 \
  --max-queued-requests 64 \
  --chunked-prefill-size 8192 \
  --enable-priority-scheduling \
  --default-priority-value 0 \
  --enable-metrics
```

The less stable setting that usually fails within a few hours is:

```bash
--speculative-num-steps 3 \
--speculative-eagle-topk 1 \
--speculative-num-draft-tokens 4
```

## Observed logs

```text
[full] total=464256, available=768, evictable=432960, protected=30464, session_held=0, uncached=0
[2026-05-26 13:36:13 TP5] Pyspy failed (py-spy dump --native --pid 26818). Error: Permission Denied
[2026-05-26 13:36:14 TP5] All pyspy dump attempts failed for PID 26818.
[2026-05-26 13:36:14 TP5] Scheduler watchdog timeout (self.watchdog_timeout=300, self.soft=False)
[2026-05-26 13:36:19] SIGQUIT received. signum=None, frame=None. It usually means one child failed.
```

Earlier run with more aggressive EAGLE settings also showed NCCL collective mismatch / timeout symptoms:

```text
Rank 4: all_gather_into_tensor -> logits_processor -> deepseek_nextn -> eagle_worker_v2
Rank 6: broadcast -> eagle_info_v2.py:sample
WorkNCCL OpType=BROADCAST Timeout(ms)=600000
```

After changing EAGLE from `3/1/4` to `1/1/1`, the service became more stable and could run for about two days, but the HiCache/radix eviction hang still occurred.

## Expected behavior

When `available` is low but `evictable` is high, the scheduler should evict radix/HiCache entries and continue serving the request, instead of hanging until watchdog timeout.

## Environment

- Model: GLM-5.1 FP8
- TP: 8
- KV cache dtype: `fp8_e4m3`
- HiCache: enabled
- Context length: 160000
- SGLang version: v0.5.12 and several nearby versions tested
- GPU: H200-class 8 GPU node
- No recent kernel Xid/OOM errors were observed around the crash window.
